ethpm_types/utils.py

```diff
@@ -1,20 +1,75 @@
-from hashlib import md5
+from enum import Enum
+from hashlib import md5, sha3_256, sha256
 
 
-def compute_checksum(source: bytes, algorithm: str = "md5") -> str:
-    if algorithm == "md5":
-        hasher = md5
-    else:
-        raise Exception("Unknown algorithm")
+class Algorithm(str, Enum):
+    MD5 = "md5"
+    SHA3 = "sha3"
+    SHA256 = "sha256"
 
-    return hasher(source).hexdigest()
 
+def is_valid_hex(data: str) -> bool:
+    if not data.startswith("0x"):
+        return False
 
-def is_valid_hash(data: str) -> bool:
-    if set(data.lower()) > set("1234567890abcdef"):
+    if set(data[2:].lower()) > set("1234567890abcdef"):
         return False
 
     if len(data) % 2 != 0:
         return False
 
     return True
+
+
+class Hex(str):
```

Comment:
Would this class be useful outside of ethpm-types?
Like we do need our own "eth-base-types` repository?

Reply:
Maybe eventually

Reply:
Okay punting to #18